src/cli/commands/test/iac-local-execution/results-formatter.ts

```diff
@@ -41,15 +41,22 @@ const engineTypeToProjectType = {
 function formatScanResult(
   scanResult: IacFileScanResult,
   meta: TestMeta,
-  { severityThreshold, json, sarif }: IaCTestFlags,
+  {
+    severityThreshold,
+    json,
+    sarif,
+    'json-file-output': jsonFileOutput,
+    'sarif-file-output': sarifFileOutput,
+  }: IaCTestFlags,
 ): FormattedResult {
   const formattedIssues = scanResult.violatedPolicies.map((policy) => {
     const cloudConfigPath =
       scanResult.docId !== undefined
         ? [`[DocId:${scanResult.docId}]`].concat(policy.msg.split('.'))
         : policy.msg.split('.');
 
-    const shouldExtractLineNumber = json || sarif;
+    const shouldExtractLineNumber =
+      json || sarif || jsonFileOutput || sarifFileOutput;
```

Comment:
nit: Would it be better to have an array containing the flags and check each one?
```suggestion
    const flagsRequiringLineNumber = ['json', 'sarif', ...]
    const shouldExtractLineNumber = flagsRequiringLineNumber.some(flag => options[flag]);
```